src/device/device_ledger.cpp

```diff
@@ -1209,20 +1199,13 @@ namespace hw {
         hw::ledger::check32("derive_secret_key", "derived_sec", derived_sec_x.data, derived_sec_clear.data);
         #endif
 
-        unlock_device();
-      }catch (...)  {
-        unlock_device();
-        throw;
-      }
-      return true;
+        return true;
     }
 
     bool device_ledger::derive_public_key(const crypto::key_derivation &derivation, const std::size_t output_index, const crypto::public_key &pub, crypto::public_key &derived_pub){
```

Comment:
Doesn't this function also require the same branch logic with `has_view_key`?

Reply:
No (if I correctly analyse the code). Once the TX identified, the whole process is done throw device, even the re-computation of derivation encrypted from device